src/tiered/tiered_handle.c

```diff
@@ -8,6 +8,64 @@
 
 #include "wt_internal.h"
 
+/*
+ * __tiered_name_check --
+ *     Check if the given tiered table name exists in the shared storage. This is called from create
+ *     to check if this name had previously existed, was dropped and we're now trying to create the
+ *     same name. We cannot do that if shared objects exist for the old table.
```

Comment:
I had a long thought about naming and uniqueness, that I wrote up in WT-7702.
Still, I think the approach you're using is fine for now.